service/src/ConstConfigIOGroup.cpp

```diff
@@ -203,13 +209,21 @@ namespace geopm
                             GEOPM_ERROR_INVALID, __FILE__, __LINE__);
         }
         else if (domain_idx < 0 ||
-                 static_cast<std::size_t>(domain_idx) >= it->second->values.size()) {
+                 domain_idx >= m_platform_topo.num_domain(domain_type)) {
             throw Exception("ConstConfigIOGroup::read_signal(): domain_idx " +
                             std::to_string(domain_idx) + " out of range.",
                             GEOPM_ERROR_INVALID, __FILE__, __LINE__);
         }
 
-        return it->second->values[domain_idx];
+        double value = 0.0;
```

Comment:
Other IOGroups use NAN as a default value as a precaution; if we get to the return without setting the value the NAN indicates something has gone wrong.  Should we follow the same pattern here?

Reply:
Yeah, that makes sense. I'll make that change.